File: 03 - frontend/react-library/src/models/AddProjectRequest.ts
class AddProjectRequest{
    title: string;
    description: string;
    start_date: Date; 
    location: string;
    project_type: string;
    weather_conditions: string;
    expected_duration: number;
    project_manager: string;
    contact_email: string;
    contact_phone: string;
    images? : string[]
    
    constructor(title: string, project_manager: string, description: string,
        location: string, project_type: string, weather_conditions: string,
        expected_duration: number, contact_email: string, contact_phone: string,
        start_date: Date){
        this.title = title;
        this.description = description;
        this.start_date = start_date;
        this.location = location;
        this.project_type = project_type;
        this.weather_conditions = weather_conditions;
        this.expected_duration = expected_duration;
        this.project_manager = project_manager;
        this.contact_email = contact_email;
        this.contact_phone = contact_phone;
    }
}
export default AddProjectRequest;

File: 02 - backend/spring-boot-Inspection-RestfulWebservices/src/main/java/com/mkr/Inspection/requestModels/AddProjectRequest.java
package com.mkr.Inspection.requestModels;

import lombok.Data;
import java.util.Date;
@Data
public class AddProjectRequest {
    private String title;
    private String description;
    private Date start_date;
    private String location;
    private String project_type;
    private String weather_conditions;
    private int expected_duration;
    private String project_manager;
    private String contact_email;
    private String contact_phone;
    private String[] images;

}
